scripts/linkbot/monitor-ub-server.js

```diff
@@ -28,27 +28,42 @@ JSDOM.fromURL(url, {}).then(dom => {
         .filter(ref => ref.textContent.match(reg))
         .filter(ref => !ref.textContent.match(reg).groups.minor) // Ignore the minor number
         .reduce((acc, ref) => {
-            acc.push(ref.textContent.match(reg))
-            return acc
+            acc.push(ref.textContent.match(reg));
+            return acc;
         }, [])
         .sort((a,b) => {
-            return parseInt(b.groups.release) - parseInt(a.groups.release)
-        })
-    var matchOn = matches[0].input
+            return parseInt(b.groups.release) - parseInt(a.groups.release);
+        });
+    var matchOn = matches[0].input;
 
     return matchOn;
 
     // Get the release image url from the releases (sub)-page
     // const url = "http://cdimage.ubuntu.com/ubuntu/releases/"
 }).then(releaseVersion => {
-    var releaseVersion = releaseVersion.replace(/\s/g, "").replace(/\//g, "")
-    JSDOM.fromURL(`${url}${releaseVersion}/release/`, {}).then(dom => {
+    var releasedVersion = releaseVersion.replace(/\s/g, "").replace(/\//g, "");
+    JSDOM.fromURL(`${url}${releasedVersion}/release/`, {}).then(dom => {
         var document = dom.window.document;
         var refs = document.getElementsByTagName("a");
-        const match = Array.from(refs).find(ref => ref.href.match(`.*ubuntu-${releaseVersion}\.?[0-9]+-preinstalled-server-armhf.*\.img\.xz$`))
+        const match = Array.from(refs).find(ref => ref.href.match(`.*ubuntu-${releasedVersion}\.?[0-9]+-preinstalled-server-armhf.*\.img\.xz$`));
         if (match) {
-            console.log(`Ubuntu server image has a new release: ${match}`)
-            updateURLLink(`${target}=\"${match}\"`, target)
+            console.log(`Ubuntu server image has a new release: ${match}`);
+            updateURLLink(`${target}=\"${match}\"`, target);
         }
     })
-});
+        .catch(err => {
+            console.log('Failed to get the update URL');
+            console.log(`${url}${releaseVersion}/release/`);
+            JSDOM.fromURL(`${url}${releasedVersion}/beta/`, {}).then(dom => {
+                console.log('Only the beta is out still');
+                process.exit(0);
+            });
+            process.exit(1);
```

Comment:
This I could imagine to cause problems... as the process would exit with 1 once the `fromURL` call was issued, and then the `exit(0)` might never reach anyone...
I think an option to get this done might be:
```
let exitCode = 0;
JSDOM.fromURL(...)
  .then(dom => {
    ...
    return Promise.resolve();
  })
  .catch(err => {
    ...
    exitCode = 1;
    return JSDOM.fromURL(...)
  })
  .then(dom => {
    ...
    exitCode = 0;
  })
  .finally(() => process.exit(exitCode));
```

Reply:
Could another idea here be to remove the `exits`, and simply catch the ones I want to not fail, and re-raise in the ones I want to fail ?

Reply:
absolutely - just throw whenever and make the crash exit the process 👍

Reply:
Done :)